src/liballoc_system/lib.rs

```diff
@@ -19,7 +19,7 @@
             issue = "27783")]
 #![feature(allocator)]
 #![feature(staged_api)]
-#![cfg_attr(unix, feature(libc))]
+#![cfg_attr(any(unix, target_os = "redox"), feature(libc))]
```

Comment:
For `alloc_system`, we simply reuse the `unix` setup to allow a `libc` implementation of an allocator